File: app/src/main/java/app/consulto/extras/chats.java
package app.consulto.extras;

import android.content.Intent;
import android.graphics.Bitmap;
import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.util.LruCache;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.android.volley.toolbox.ImageLoader;

import org.jetbrains.annotations.NotNull;

import app.consulto.R;
import app.consulto.TalkActivity;
import app.consulto.griddetails;
import app.consulto.gridet;
import app.consulto.grids;

public class chats extends Fragment
{
    private View view;
    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        view = getLayoutInflater().inflate(R.layout.chats,container,false);
        get_chats();
        return view;
    }
    private void get_chats()
    {
        grids.imageLoader = new ImageLoader(grids.queue, new ImageLoader.ImageCache() {
            LruCache<String, Bitmap> cahce=  new LruCache<String,Bitmap>(grids.mysubscriptions.length);
            @Override
            public Bitmap getBitmap(String url) {
                return cahce.get(url);
            }

            @Override
            public void putBitmap(String url, Bitmap bitmap) {
                cahce.put(url,bitmap);
            }
        });
        final Intent i = new Intent(grids.con, TalkActivity.class);
        Handler h = new Handler()
        {
            @Override
            public void handleMessage(@NonNull Message msg) {
                super.handleMessage(msg);
                i.putExtras(msg.getData());
                startActivity(i);
            }
        };
        grids.bun = new Bundle[grids.mysubscriptions.length];
        ListView ls = (ListView)view.findViewById(R.id.chat_list);
        grids.c = new mysubsadapter(grids.con,R.layout.enrolled_listview,grids.mysubscriptions,h,grids.imageLoader,grids.bun);
        get_bundles();
        ls.setAdapter(grids.c);
    }
    private void get_bundles()
    {
        gridet ob = new gridet();
        ob.gri = new griddetails() {
            @Override
            public void griddetails(@NotNull String name, @NotNull String dess, @NotNull String rating, @NotNull String[][] reviews) {
                Bundle b = new Bundle();
                b.putString("title",name);
                b.putString("id", String.valueOf(grids.buncount));
                b.putString("email", grids.email);
                b.putString("password", grids.password);
                b.putString("pubkey", grids.pubkey);
                b.putString("privkey", grids.privkey);
                grids.bun[grids.buncount-1] = b;
                grids.buncount++;
                if(grids.buncount<=grids.mysubscriptions.length)
                {
                    get_bundles();
                }
                else
                {
                    grids.c.notifyDataSetChanged();
                    grids.buncount = 1;
                }
            }
        };
        ob.get_details(String.valueOf(grids.buncount),grids.email,grids.password, grids.queue,grids.con);
    }
}

File: app/src/main/java/app/consulto/seller/chats.kt
package app.consulto.seller

import android.app.AlarmManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Message
import android.view.*
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import app.consulto.R
import com.android.volley.RequestQueue
import com.android.volley.toolbox.Volley
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener
import java.io.File
import java.util.*

class chats : AppCompatActivity() {
    private var id = 0
    private var title = ""
    private var email = ""
    lateinit var queue: RequestQueue
    lateinit var c:listadapt
    lateinit var firebaseDatabase:FirebaseDatabase
    private var password= ""
    lateinit var users:List<String>
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_chats)
        setSupportActionBar(findViewById(R.id.toolbar))
        val bun = intent.extras
        queue = Volley.newRequestQueue(this)
        id = bun!!.getInt("id")
        title = bun.getString("title").toString()
        email = bun.getString("email").toString()
        password = bun.getString("password").toString()
        supportActionBar!!.setTitle(title)
       get_live_users()
        val ser = Intent(this,customer_service::class.java)
       startService(ser)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
      menuInflater.inflate(R.menu.seller_menu,menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        logout()
        return true
    }
    fun logout() {
        File(filesDir, "username.txt").delete()
        File(filesDir, "password.txt").delete()
        finish()
    }
    private fun make_list()
    {
        val i = Intent(this,SellerTalk::class.java)
        val h = object:Handler()
        {
            override fun handleMessage(msg: Message) {
                super.handleMessage(msg)
                val bun  =Bundle();
                bun.putString("email",email)
                bun.putString("password",password)
                bun.putString("id",id.toString())
                bun.putString("to",msg.data.getString("user"))
                i.putExtras(bun)
                startActivity(i)
            }
        }
        val listview = findViewById<ListView>(R.id.listvoew);
        c = listadapt(this,R.layout.live_user_listview,users,h)
        listview.adapter = c;
    }
    private fun get_live_users()
    {
     firebaseDatabase = FirebaseDatabase.getInstance()
        val ref = firebaseDatabase.getReference("live"+id)
        ref.addValueEventListener(object:ValueEventListener
        {
            override fun onCancelled(error: DatabaseError) {

            }

            override fun onDataChange(snapshot: DataSnapshot) {
                if(snapshot.getValue()!=null)
                {
                    val usrs = snapshot.getValue().toString();
                    users = usrs.split(",")
                    make_list()
                }
            }
        })
    }
}
class listadapt(con: Context,res:Int,usrs:List<String>,h:Handler) : ArrayAdapter<String>(con,res)
{
    lateinit var handler: Handler
    lateinit var users:List<String>
    init {
        this.handler = h
        this.users = usrs
    }
    override fun getCount(): Int {
        return users.size-1
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val v:View = LayoutInflater.from(context).inflate(R.layout.live_user_listview,null)
        v.findViewById<TextView>(R.id.user).setText("User "+(position+1))
        v.setOnClickListener(object :View.OnClickListener
        {
            override fun onClick(v: View?) {
                val b = Bundle();
                val m = Message();
                b.putString("user",users[position])
                m.data = b
                handler.sendMessage(m)
            }
        })
        return v;
    }
}